mfrc522/iso14443com.py

```diff
@@ -0,0 +1,55 @@
+import crcmod
+from .mfrc522 import *
+import struct
+
+SEL_CASCADE_1   = 0x93
+SEL_CASCADE_2   = 0x95
+SEL_CASCADE_3   = 0x97
+
+__crc_func = crcmod.mkCrcFun(0x11021, initCrc=0x6363)
+
+def __calculate_crc(data):
+    return struct.pack('<H', __crc_func(data))
+
+def __perform_cascade(module, cascade_level):
```

Comment:
The recursive version would make sense if anticollision were implemented. But for now, what about:
```python
CASCADE_LEVELS = [ 0x93, 0x95, 0x97 ]

def __perform_cascade(rfid_module):
    result = b''
    for cascade_level in CASCADE_LEVELS:
        # transmit ANTICOLLISION command
        uid_cln = module.transcieve(bytes([cascade_level, 0x20]))  # BTW, that's spelled "transceive".
        if len(uid_cln) != 4: raise ...  # Don't ignore errors!

        # transmit SELECT command
        data = bytes((cascade_level, 0x70)) + bytes(uid_cln)
        data += __calculate_crc(data)
        response = rfid_module.transcieve(data)

        result += uid_cln[1:4]

        if response[0] & 0x24 in [0x0, 0x20]:
            return result
        elif response[0] & 0x04:
            pass  # continue the cascade
        else:
            # Something's wrong if we get here.
            raise ...  # Say so.
```